tst/EnergyPlus/unit/AirflowNetworkBalanceManager.unit.cc

```diff
@@ -8210,4 +8212,34 @@ TEST_F(EnergyPlusFixture, BasicAdvancedSingleSided)
     }
 }
 
+TEST_F(EnergyPlusFixture, AFN_CheckNumOfFansInAirLoopTest)
+{
+    // #6748
+    DataAirSystems::PrimaryAirSystem.allocate(1);
+    DataAirSystems::PrimaryAirSystem(1).NumBranches = 1;
+    DataAirSystems::PrimaryAirSystem(1).Branch.allocate(1);
+    DataAirSystems::PrimaryAirSystem(1).Branch(1).TotalComponents = 3;
+    DataAirSystems::PrimaryAirSystem(1).Branch(1).Comp.allocate(3);
+    DataAirSystems::PrimaryAirSystem(1).Branch(1).Comp(1).TypeOf = "Fan:ConstantVolume";
+    DataAirSystems::PrimaryAirSystem(1).Branch(1).Comp(2).TypeOf = "Fan:VariableVolume";
+    DataAirSystems::PrimaryAirSystem(1).Branch(1).Comp(1).Name = "CVF";
+    DataAirSystems::PrimaryAirSystem(1).Branch(1).Comp(2).Name = "VAV";
+
+    ASSERT_THROW(ValidateDistributionSystem(), std::runtime_error);
+
+    std::string const error_string = delimited_string({
+        "   ** Severe  ** ValidateDistributionSystem: An AirLoop branch, , has two or more fans: CVF,VAV",
+        "   **   ~~~   ** The AirflowNetwork model allows a single supply fan in an AirLoop only. Please make "
+        "changes in the input file accordingly.",
+        "   **  Fatal  ** ValidateDistributionSystem: Program terminates for preceding reason(s).",
+        "   ...Summary of Errors that led to program termination:",
+        "   ..... Reference severe error count=1",
+        "   ..... Last severe error=ValidateDistributionSystem: An AirLoop branch, , has two or more fans: CVF,VAV",
+        });
+
+    EXPECT_TRUE(compare_err_stream(error_string, true));
+
+    DataAirSystems::PrimaryAirSystem.deallocate();
```

Comment:
Cleanup like this should be done in the `clear_state` function in `DataAirSystems.cc`, not here.

Reply:
@Myoldmopar The clear_state function in DataAirSystems.cc has the deallocate statement: PrimaryAirSystem.deallocate(). I am going to remove the deallocate statement in the unit test file.